zerver/lib/email_mirror.py

```diff
@@ -171,6 +173,11 @@ class ZulipEmailForwardUserError(ZulipEmailForwardError):
     pass
 
 def send_zulip(sender: UserProfile, stream: Stream, topic: str, content: str) -> None:
+    if (len(topic) > MAX_TOPIC_NAME_LENGTH):
+        if not (messages_for_topic(stream.recipient_id, topic).filter(
```

Comment:
Given that `topic` in this line is already guaranteed to be over the max length, `messages_for_topic(..., topic)` can't return anything. It should be looking for truncated version of the topic.

Reply:
We should also have tests for these scenarios - verify that the `Subject: ...` line is inserted into the content correctly based on whether an existing, matching topic was found or not.

Reply:
Got it, I'm on it!